config/initializers/sidekiq_hacks.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+# This monkey path can be deleted after migrating to Sidekiq >= 7
+module OptionsPatch
+  def client_opts(options)
+    valid_keys = %i[url host port namespace sentinels role path username password ssl_params]
+    super(options.slice(*valid_keys))
+  end
+end
+
+Sidekiq::RedisConnection::RedisAdapter.prepend OptionsPatch
+
+#--------------------------
+
+# Delete this after solving https://github.com/breamware/sidekiq-batch/issues/86
+module CorrectArgumentBatch
+  def enqueue_callbacks(event, bid)
+    event_name = event.to_s
+    batch_key = "BID-#{bid}"
+    callback_key = "#{batch_key}-callbacks-#{event_name}"
+    already_processed, _, callbacks, queue, parent_bid, callback_batch = Sidekiq.redis do |r|
+      r.multi do |pipeline|
+        pipeline.hget(batch_key, event_name)
+        pipeline.hset(batch_key, event_name, true.to_s)
```

Comment:
`sidekiq-batch` doesn't support the `redis-client` gem either, but we need the last to add support for TLS, so I had to add this patch and overwrite the whole method only to change this line, to include the call to `.to_s`.

I created an issue but I don't think `sidekiq-batch` is maintained anymore. It's a problem because Sidekiq 7 only supports `redis-client` so `sidekiq-batch` is not really compatible with that version.

Also, I can't know in which other places is `sidekiq-batch` sending wrong parameters to redis. So there will be potential issues when deploying.

Reply:
You can also create a PR in `sidekiq-batch` should be easy enough and more likely to attract any surviving maintainer's  attention.

Reply:
I removed this monkey patch and created a fork which includes this and another change. I'm not sure how many other changes will I have to make on `sidekiq-batch` to make it support `redis-client`, so I prefer to not keep adding patches.

https://github.com/3scale/sidekiq-batch